codice/ddf
# DDF-04291 Removed org.codice.ddf.ui.searchui.filter.RedirectServlet.config in favor of setting the property default in the class

#### What does this PR do?
This PR
- Removes _org.codice.ddf.ui.searchui.filter.RedirectServlet.config_ in favor of setting the property defaults in the class. See https://github.com/codice/ddf/issues/4291 for motivation.
- Injects the `defaultUri` String directly instead of injecting a `RedirectConfiguration` to wrap the `defaultUri` String
- Updates the default `defaultUri` to a relative URI instead of absolute to simplify the default value and demonstrate that a relative URI can be specified
#### Who is reviewing it? 
@stustison @jhunzik @Kjames5269 
#### Ask 2 committers to review/merge the PR and tag them here.
@brendan-hofmann
@vinamartin
#### How should this be tested?
`log:set DEBUG org.codice.ddf.ui.searchui.filter.RedirectServlet`
- Verify that the `org.codice.ddf.ui.searchui.filter.RedirectServlet` line under `Search UI` app -> `Configuration` -> `Search UI Redirect` in the Admin Console is not displayed on a fresh install.
  Before:
  ![image](https://user-images.githubusercontent.com/8041246/52020614-334be780-24af-11e9-83de-744c363ea2e7.png)
  After:
  ![image](https://user-images.githubusercontent.com/8041246/53755985-33a41d80-3e75-11e9-9301-a72b2137b06c.png)
- Confirm that `/search` redirects to `/search/catalog.`
- Confirm that the `Redirect URI` input box is filled with `${org.codice.ddf.external.context}/search/catalog/` by default for `Search UI` app -> `Configuration` -> `Search UI Redirect` in the Admin Console. 
- Confirm that saving the default URI in the Admin Console succeeds.
- Confirm that Search UI Redirect can be configured to absolute or relative URIs.
- Bonus: Follow the test steps in https://github.com/codice/ddf/pull/3309 to confirm that the redirect works after configuring a context path. `${org.codice.ddf.external.context}/search/catalog/` is the relative URI for `${org.codice.ddf.external.protocol}${org.codice.ddf.external.hostname}:${org.codice.ddf.external.port}${org.codice.ddf.external.context}/search/catalog/`.
#### Any background context you want to provide?
See https://github.com/codice/ddf/pull/3309 for details on `${org.codice.ddf.external.context}`.
#### What are the relevant tickets?

https://github.com/codice/ddf/issues/4291
#### Screenshots

![image](https://user-images.githubusercontent.com/8041246/52021164-31832380-24b1-11e9-830a-95461345ff7c.png)
![image](https://user-images.githubusercontent.com/8041246/53756017-546c7300-3e75-11e9-955c-be08c94d4f6e.png)
![image](https://user-images.githubusercontent.com/8041246/53756086-7ebe3080-3e75-11e9-9253-49972c0e6310.png)
#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
